### PR TITLE
improvement(upgrade): retry query execute

### PR DESCRIPTION
**Pull request by juliayakovlev**
Issue: https://github.com/scylladb/scylla-enterprise/issues/3820 Workaround for this issue - try to retry queries if it failed

### Testing

- [ ] https://jenkins.scylladb.com/job/enterprise-2024.1/job/reproducers/job/rolling-upgrade-azure-image-test/2/

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevent to this change (if needed)

### CODE REVIEW

File: sdcm/fill_db_data.py [hunk 1]
```diff
@@ -3309,8 +3306,13 @@ def run_db_queries(self, session, default_fetch_size):
                             item["cdc_tables"][cdc_table] = self.get_cdc_log_rows(session, cdc_table)
                             LOGGER.debug(item["cdc_tables"][cdc_table])
 
+    @staticmethod
+    @retrying()
+    def run_query_with_retry(session, query):
```
Comment (fruch) [inline sdcm/fill_db_data.py:3311]: I would add `**kwargs`, so we can add things like timeout and such to the `execute`

File: sdcm/fill_db_data.py [hunk 2]
```diff
@@ -3309,8 +3306,13 @@ def run_db_queries(self, session, default_fetch_size):
                             item["cdc_tables"][cdc_table] = self.get_cdc_log_rows(session, cdc_table)
                             LOGGER.debug(item["cdc_tables"][cdc_table])
 
+    @staticmethod
+    @retrying()
```
Comment (fruch) [inline sdcm/fill_db_data.py:3310]: I would add the exact number of retries here, so it would clearer to read.

File: sdcm/fill_db_data.py [hunk 3]
```diff
@@ -3237,29 +3235,28 @@ def _run_db_queries(self, item, session):
         for i in range(len(item['queries'])):
             try:
                 if item['queries'][i].startswith("#SORTED"):
-                    res = session.execute(item['queries'][i].replace('#SORTED', ''))
+                    res = self.run_query_with_retry(session=session, query=item['queries'][i].replace('#SORTED', ''))
```
Comment (fruch) [inline sdcm/fill_db_data.py:3238]: I'm not sure we want retries for all of the `_run_db_queries`, we don't usually have issues with those, and if they won't work at first shot, we do want to know about them

File: sdcm/fill_db_data.py (hunk 1 above)
Comment (juliayakovlev) [reply sdcm/fill_db_data.py]: done

Comment (juliayakovlev) [reply sdcm/fill_db_data.py]: done

File: sdcm/fill_db_data.py (hunk 2 above)
Comment (juliayakovlev) [reply sdcm/fill_db_data.py]: done

File: sdcm/fill_db_data.py (hunk 3 above)
Comment (juliayakovlev) [reply sdcm/fill_db_data.py]: done

Comment (roydahan) [reply sdcm/fill_db_data.py]: @fruch its a good question if we want to know about them or not.
Originally we wanted and reported some of them, but looks like the approach is to ask first for retries.
We can ask in one of the existing issues or in a new issue or in Slack engineering what should be the default behavior.

Comment (fruch) [reply sdcm/fill_db_data.py]: We properly want to fix the driver and have support of exponential backoffs, and set it per session 

https://github.com/scylladb/python-driver/issues/91

**Comment by kbr-scylla** [comment]
But this may mask a real issue, I think we need to investigate https://github.com/scylladb/scylla-enterprise/issues/3820 to find root cause first (sorry it's taking so long, we're mostly focusing on delivering raft-topology now)

**Pull request closed without merging**